server.py

```diff
@@ -15,15 +15,10 @@ def main():
             except socket.error, e:
                 print "Client aborted "+str(e)
                 break
-            if data == "SHUTDOWN":
-                conn.sendall("GOODBYE")
-                break
-            else:
-                print data
-                conn.send(data + data)
+            print data
+            conn.send(data + data)
```

Comment:
If the client shuts down its connection, you're sending an empty string to a closed connection, I think this might throw an exception?

Reply:
fixed